src/gennylib/src/context.cpp

```diff
@@ -133,15 +143,18 @@ DefaultRandom& WorkloadContext::getRNGForThread(ActorId id) {
     if (this->isDone()) {
         BOOST_THROW_EXCEPTION(std::logic_error("Cannot create RNGs after setup"));
     }
-    if (auto rng = _rngRegistry.find(id); rng == _rngRegistry.end()) {
-        auto [it, success] = _rngRegistry.try_emplace(id, _rng());
-        if (!success) {
-            // This should be impossible.
-            // But invariants don't hurt we only call this during setup
-            throw std::logic_error("Already have DefaultRandom for Actor " + std::to_string(id));
-        }
+
+    if (id < 1) {
+        BOOST_THROW_EXCEPTION(std::logic_error("ActorId must be 1 or greater."));
     }
-    return _rngRegistry[id];
+
+    // Create RNGs for all ids up to our own.
```

Comment:
does it make sense to eagerly do this when requesting the actorIds? Keeps from having to have another lock. Not sure of all pros/cons of this, but if it works out, I think make a DefaultRandom a `protected` member of `genny::Actor` similar to its `_id` field.

Reply:
That's a good idea! In an earlier iteration of this I didn't want to, due to the producers having all the knowledge of how many actors there will be, but pre-claiming the IDs changes that.